docs/arguments.md

```diff
@@ -92,6 +92,43 @@ The ingester query API was improved over time, but defaults to the old behaviour
 - `-distributor.extra-query-delay`
    This is used by a component with an embedded distributor (Querier and Ruler) to control how long to wait until sending more than the minimum amount of queries needed for a successful response.
 
+- `distributor.accept-ha-samples`
+   Per-user flag to enable handling of samples with external labels for identifying replicas in an HA Prometheus setup. This defaults to false, and is technically defined in the Distributor limits.
+
+- `distributor.enable-ha-tracker` 
+   Enable the distributors HA tracker so that it can accept samples from Prometheus HA replicas gracefully (requires labels). Global (for distributors), this ensures that the necessary internal data structures for the HA handling are created. The option `accept-ha-samples` is still needed to enable ingestion of HA samples per user ID until that flag is defaulted to true.
+
+### HA Tracker
+
+HA tracking has two of it's own flags:
+- `ha-tracker.cluster`
+   Prometheus label to look for in samples to identify a Prometheus HA cluster. (default "cluster")
+- `ha-tracker.replica`
+   Prometheus label to look for in samples to identify a Prometheus HA replica. (default "__replica__")
+
+It's reasonable to assume people probably already have a `cluster` label, or something similar. If not, they should add one along with `__replica__`
+via external labels in their Prometheus config.
+
+HA Tracking looks for the two labels (which can be overwritten per user)
+
+It also talks to a KVStore and has it's own copies of the same flags used by the Distributor to connect to for the ring.
+- `ha-tracker.consul.acltoken`
+   ACL Token used to interact with Consul.
+- `ha-tracker.consul.client-timeout`
+   HTTP timeout when talking to consul (default 20s)
+- `ha-tracker.consul.consistent-reads`
+   Enable consistent reads to consul. (default true)
+- `ha-tracker.consul.hostname`
+   Hostname and port of Consul. (default "localhost:8500")
+- `ha-tracker.consul.prefix`
+   Prefix for keys in Consul. Should end with a /. (default "collectors/")
```

Comment:
Perhaps add a not here to set this value to something other than the default if you are using the same consul as for the ring. Or change the default to something like `collectors/HA/`?

Reply:
Hmm. I don't think _not_ changing the value to something other than the prefix used for the ring would cause any issues. The keys for HA tracking are `user ID/cluster ID`.

Unfortunately without some more refactoring we wouldn't be able to change the default. We just call the `RegisterFlags` function for the other config structs, passing an optional prefix.

Reply:
I don't think it will cause issues unless you happen to have a tenant named `ring`. Mostly it was nice to debug that everything was looking correct for them to be in their own folder.

Reply:
Agreed. Best practice imo would be to use different prefixes.